training_config/transformer_squad.bert-base.jsonnet

```diff
@@ -0,0 +1,60 @@
+local transformer_model = "bert-base-cased";
+local epochs = 3;
+local batch_size = 8;
+
+// This defines the number of instances, not the number of questions. One question can end up as
+// multiple instances.
+local number_of_train_instances = 87825;
+local number_of_dev_instances = 10826;
+
+{
+    "dataset_reader": {
+        "type": "transformer_squad",
+        "transformer_model_name": transformer_model,
+        "skip_invalid_examples": true,
+        //"max_questions": 100  // debug setting
+    },
+    "validation_dataset_reader": self.dataset_reader + {
+        "skip_invalid_examples": false,
+    },
+    "train_data_path": "https://allennlp.s3.amazonaws.com/datasets/squad/squad-train-v1.1.json",
+    "validation_data_path": "https://allennlp.s3.amazonaws.com/datasets/squad/squad-dev-v1.1.json",
+    "model": {
+        "type": "transformer_qa",
+        "text_field_embedder": {
+            "token_embedders": {
+                "tokens": {
+                    "type": "pretrained_transformer",
+                    "model_name": transformer_model
+                },
+            },
+        },
+    },
+    "iterator": {
+        "type": "basic",
```

Comment:
I'm a bit surprised to see `basic` here. Generally `bucket` will result in more throughput. Is this intentional?

Reply:
I'll try it and see if it changes scores. I probably used `basic` because Bidaf used `basic`.

Reply:
Ah! In that case there's likely some reason why it was chosen initially. Maybe @matt-gardner knows?

Reply:
BiDAF doesn't use it: https://github.com/allenai/allennlp-reading-comprehension/blob/fa60af5736a22455d275e663d3dd1ecc838e400c/training_config/bidaf.jsonnet#L82-L86.  Our test fixtures often do, though, and that seems like a more likely source for this.

Reply:
If it's not from bidaf, it must be from Oyvind's version of this model. Anyways, the point is I just copied this over, I didn't try it. It's running now, and if `bucket` is no worse than `basic`, I'll switch to that.

Reply:
`bucket` loses 0.2% F1, and trains 30% faster, so it goes in: bb038a2a08d6e15c20e3d9131f373e10a98c0f90